src/sms-command/FormKeyValueParser/useUpdateCommandMutation.js

```diff
@@ -77,7 +77,30 @@ const updateKeyValueParserMutation = {
 export const useUpdateCommandMutation = ({ id, onAfterChange }) => {
     const engine = useDataEngine()
     const onSubmit = (values) => {
-        const variables = { ...values, id }
+        const variables = {
+            ...values,
+            id,
+            smsCodes:
+                !values.smsCodes || !Object.values(values.smsCodes).length
+                    ? []
+                    : Object.entries(values.smsCodes).map(([key, value]) => {
+                          const [dataElementId, categoryOptionComboId] =
+                              key.split('-')
+
+                          return {
+                              ...(categoryOptionComboId
+                                  ? {
+                                        optionId: { id: categoryOptionComboId },
+                                    }
+                                  : {}),
+                              code: value,
+                              dataElement: {
+                                  id: dataElementId,
+                              },
+                          }
+                      }),
+        }
```

Comment:
Same as above, seems like this could've been extracted to a utility-function. Understand if this is due to time-pressure and want to get it in regardless though.